src/common/ServerCompatBanner.js

```diff
@@ -0,0 +1,101 @@
+/* @flow strict-local */
+
+import React from 'react';
+import { View } from 'react-native';
+import { SafeAreaView } from 'react-native-safe-area-context';
+
+import store from '../boot/store';
+import { createStyleSheet, HALF_COLOR } from '../styles';
+import { useSelector, useDispatch } from '../react-redux';
+import Label from './Label';
+import { getActiveAccount } from '../account/accountsSelectors';
+import { getIsAdmin, getSession } from '../directSelectors';
+import { dismissCompatNotice } from '../session/sessionActions';
+import ZulipTextButton from './ZulipTextButton';
+import { openLinkWithUserPreference } from '../utils/openLink';
+
+const styles = createStyleSheet({
+  wrapper: {
+    backgroundColor: HALF_COLOR,
+    padding: 8,
+  },
+  textRow: {
+    padding: 8,
+    flexDirection: 'row',
+  },
+  buttonsRow: {
+    flexDirection: 'row',
+    justifyContent: 'flex-end',
+  },
+});
+
+type Props = $ReadOnly<{||}>;
+
+/**
+ * A "nag banner" saying that the server version is unsupported.
+ *
+ * Currently just checks if it's less than 2.1.0, since that's the
+ * marker that was put down in zulip/zulip@d7d79b57b. In the future,
+ * this won't have to be so hard-coded; we may use a timer or
+ * something.
+ */
+// Made with somewhat careful attention to
+// https://material.io/components/banners. Please consult that before
+// making layout changes, and try to make them in a direction that
+// brings us closer to those guidelines.
+export default function ServerCompatBanner(props: Props) {
+  const dispatch = useDispatch();
+  const hasDismissedServerCompatNotice = useSelector(
+    state => getSession(state).hasDismissedServerCompatNotice,
+  );
+  const zulipVersion = useSelector(state => getActiveAccount(state).zulipVersion);
+  const realm = useSelector(state => getActiveAccount(state).realm);
+  const isAdmin = useSelector(getIsAdmin);
+
+  if (zulipVersion && !zulipVersion.isAtLeast('2.1.0') && !hasDismissedServerCompatNotice) {
+    return (
+      <SafeAreaView mode="padding" edges={['right', 'left']} style={styles.wrapper}>
+        <View style={styles.textRow}>
+          {isAdmin ? (
+            <Label
+              text={{
+                text:
+                  '{realm} is running Zulip Server {serverVersionRaw}, which is unsupported. Please upgrade your server as soon as possible.',
+                values: { realm: realm.toString(), serverVersionRaw: zulipVersion.raw() },
+              }}
+            />
+          ) : (
+            <Label
+              text={{
+                text:
+                  '{realm} is running Zulip Server {serverVersionRaw}, which is unsupported. Please contact your administrator about upgrading.',
+                values: { realm: realm.toString(), serverVersionRaw: zulipVersion.raw() },
+              }}
+            />
+          )}
+        </View>
+        <View style={styles.buttonsRow}>
+          <ZulipTextButton
+            leftMargin
```

Comment:
I don't think we want a left margin here. There's nothing to separate this button from except the left edge of the banner, and the banner already has its own padding.